pandas/core/frame.py

```diff
@@ -5401,6 +5401,94 @@ def _construct_result(self, result) -> "DataFrame":
         out.columns = self.columns
         return out
 
+    @Appender(
+        """
+Returns
+-------
+DataFrame
+    DataFrame that shows the differences stacked side by side.
+
+See Also
+--------
+Series.differences: Show differences.
+
+Examples
+--------
+>>> df = pd.DataFrame(
+...     {
+...         "col1": ["a", "a", "b", "b", "a"],
+...         "col2": [1.0, 2.0, 3.0, np.nan, 5.0],
+...         "col3": [1.0, 2.0, 3.0, 4.0, 5.0]
+...     },
+...     columns=["col1", "col2", "col3"],
+... )
+>>> df
+  col1  col2  col3
+0    a   1.0   1.0
+1    a   2.0   2.0
+2    b   3.0   3.0
+3    b   NaN   4.0
+4    a   5.0   5.0
+
+>>> df2 = df.copy()
+>>> df2.loc[0, 'col1'] = 'c'
+>>> df2.loc[2, 'col3'] = 4.0
+>>> df2
+  col1  col2  col3
+0    c   1.0   1.0
+1    a   2.0   2.0
+2    b   3.0   4.0
+3    b   NaN   4.0
+4    a   5.0   5.0
+
+Stack the differences on columns
+
+>>> df.differences(df2)
+  col1       col3
+  self other self other
+0    a     c  NaN   NaN
+2  NaN   NaN  3.0   4.0
+
+Stack the differences on rows
+
+>>> df.differences(df2, axis=0)
+        col1  col3
+0 self     a   NaN
+  other    c   NaN
+2 self   NaN   3.0
+  other  NaN   4.0
+
+Keep all the original indices (rows and columns)
+
+>>> df.differences(df2, keep_indices=True)
```

Comment:
Why would this output be useful? Do we need two keywords for indices and values, or can we collapse them to one?

Reply:
I think this is useful in that users can decide on their own what they want to do about the cells that are equal. For example, they may want to replace the NaNs with `-` (hyphen). This is especially useful when you want to present multiple sets of differences so you see a consistent shape in results (e.g. `df1.differences(df2)` then `df3.differences(df4)`).